FreeRTOS_Sockets.c

```diff
@@ -1599,23 +1599,42 @@ void * vSocketClose( FreeRTOS_Socket_t * pxSocket )
         FreeRTOS_Socket_t * pxOtherSocket;
         uint16_t usLocalPort = pxSocketToDelete->usLocalPort;
 
-        for( pxIterator = listGET_NEXT( pxEnd );
-             pxIterator != pxEnd;
-             pxIterator = listGET_NEXT( pxIterator ) )
+        if( pxSocketToDelete->u.xTCP.ucTCPState == ( uint8_t ) eTCP_LISTEN )
         {
-            pxOtherSocket = ipCAST_PTR_TO_TYPE_PTR( FreeRTOS_Socket_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
+            for( pxIterator = listGET_NEXT( pxEnd );
+                 pxIterator != pxEnd; )
+            {
+                pxOtherSocket = ipCAST_PTR_TO_TYPE_PTR( FreeRTOS_Socket_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
+                pxIterator = listGET_NEXT( pxIterator );
```

Comment:
This is a very minor suggestion to keep both the if and else clauses similar to each other.  Why not move this up in the for loop condition above?
`pxIterator = listGET_NEXT( pxIterator );`

i.e., the loop would become:
```
            for( pxIterator = listGET_NEXT( pxEnd );
                 pxIterator != pxEnd; 
                 pxIterator = listGET_NEXT( pxIterator )    )
```

Reply:
I don't mind this how it is. I shall approve the PR.

Reply:
Hmm, I think that would not work, because of the 
vSocketClose below I need to do the listGET_NEXT
here, the original form of the loop would access free
memory.

However, I agree that the reason why I wrote it this way
is probably not so clear as I thought it would be, so I should
definitely add a comment here, to prevent such "code-cleanup"
in the future.

Reply:
I've added a comment to hopefully make the code more clear:
"This needs to be done here, before calling vSocketClose."
Thanks for bringing this to may attention.

Reply:
Ah! Yes, I did not see the cast. Thanks for explaining that.
I have approved the PR. But it needs to be rebased before it can be merged.